### PR TITLE
[Model][LoRA]LoRA support added for idefics3

### PR DESCRIPTION
**Pull request by linitra24**
1. LoRA supported for inference in idefics3.
2. BTW, this also fixes a bug introduced in #9767, which caused the length of `dummy_data` to exceed the maximum model length setting.
3. Removed some unused code.

### CODE REVIEW

File: vllm/model_executor/models/idefics3.py [hunk 1]
```diff
@@ -357,7 +356,8 @@ def dummy_data_for_idefics3(
     max_llm_image_tokens = max_num_image_patches * image_seq_len * num_images
 
     seq_data = SequenceData.from_prompt_token_counts(
-        (hf_config.image_token_id, max_llm_image_tokens), (0, seq_len))
+        (hf_config.image_token_id, max_llm_image_tokens), 
+        (0, seq_len - max_llm_image_tokens))
```
Comment (jeejeelee) [inline vllm/model_executor/models/idefics3.py:360]: We need to handle the case where this value is negative?  See: https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/models/qwen2_vl.py#L742

**Review by jeejeelee — CHANGES_REQUESTED** [review]

File: vllm/model_executor/models/idefics3.py [hunk 2]
```diff
@@ -610,7 +606,33 @@ def forward(
 @MULTIMODAL_REGISTRY.register_max_image_tokens(get_max_idefics3_image_tokens)
 @INPUT_REGISTRY.register_dummy_data(dummy_data_for_idefics3)
 @INPUT_REGISTRY.register_input_processor(input_processor_for_idefics3)
-class Idefics3ForConditionalGeneration(nn.Module, SupportsMultiModal):
+class Idefics3ForConditionalGeneration(nn.Module, SupportsMultiModal,
```
Comment (jeejeelee) [inline vllm/model_executor/models/idefics3.py:609]: We also need to update the LoRA support documentation

File: vllm/model_executor/models/idefics3.py [hunk 3]
```diff
@@ -489,8 +487,6 @@ def _parse_and_validate_image_input(
                                             data=self._validate_pixel_values(
                                                 flatten_bn(pixel_values,
                                                            concat=True)),
-                                            rows=rows,
-                                            cols=cols,
                                             pixel_attention_mask=flatten_bn(
```
Comment (jeejeelee) [inline vllm/model_executor/models/idefics3.py:490]: Why did you delete this?

File: vllm/model_executor/models/idefics3.py (hunk 1 above)
Comment (B-201) [reply vllm/model_executor/models/idefics3.py:360]: Sounds reasonable; I'm making the changes now.

File: vllm/model_executor/models/idefics3.py (hunk 2 above)
Comment (B-201) [reply vllm/model_executor/models/idefics3.py:609]: You're right, I'll update the documentation.

File: vllm/model_executor/models/idefics3.py (hunk 3 above)
Comment (B-201) [reply vllm/model_executor/models/idefics3.py:490]: These parameters are not used.

**Review by DarkLight1337 — APPROVED** [review]
Looks good, I assume you have already tested this on your own?

**Review by jeejeelee — APPROVED** [review]
overall lgtm, @DarkLight1337  could you please  look at this PR, thanks~

**Comment by B-201** [comment]
> Looks good, I assume you have already tested this on your own?

Yes, I trained a LoRA myself and tested it locally. It runs very well, and the results align with those on transformers.
